packages/venia-concept/src/actions/cart/asyncActions.js

```diff
@@ -105,14 +105,13 @@ export const addItemToCart = (payload = {}) => {
                 }
             );
 
-            dispatch(actions.addItem.receive({ cartItem, item, quantity }));
-
             // 2019-02-07  Moved these dispatches to the success clause of
             // addItemToCart. The cart should only open on success.
             // In the catch clause, this action creator calls its own thunk,
             // so a successful retry will wind up here anyway.
             await dispatch(getCartDetails({ forceRefresh: true }));
             await dispatch(toggleDrawer('cart'));
+            dispatch(actions.addItem.receive({ cartItem, item, quantity }));
```

Comment:
Moved this action below drawer toggle, since there was a slight delay between the button going live and drawer sliding out (`getCartDetails` takes a few ms). Without the move, it still prevents the accidental double click, but there is a window of clicking again before the drawer opens.

Reply:
This is fine. I'm not sure what best practice says about dispatching actions within actions but since we're already doing it...tbh I don't think it's normal for stores to open a cart right after adding an item. What if the user wants to buy more things? Doing this is a bit opinionated of us. Sorry, just sort of rambling here. This is fine :D